packages/eslint-plugin/src/rules/no-unnecessary-type-parameters.ts

```diff
@@ -122,9 +122,7 @@ function isTypeParameterRepeatedInAST(
 
     // If the type parameter is being used as a type argument, then we
     // know the type parameter is being reused and can't be reported.
-    if (
-      reference.identifier.parent.type === AST_NODE_TYPES.TSTypeReference
-    ) {
+    if (reference.identifier.parent.type === AST_NODE_TYPES.TSTypeReference) {
```

Comment:
Hmmm, why is this changed?

Reply:
@Josh-Cena, explained [there](https://github.com/typescript-eslint/typescript-eslint/pull/9775#issuecomment-2283407308)